apps/examples/testcase/le_tc/compression/tc_compress_read.c

```diff
@@ -22,12 +22,63 @@
 #include <tinyara/config.h>
 #include <stdio.h>
 #include <stdlib.h>
+#include <errno.h>
 #include <fcntl.h>
 #include <sys/ioctl.h>
+#include <sys/types.h>
 #include <tc_common.h>
 #include <tinyara/os_api_test_drv.h>
+#include <tinyara/compression.h>
 #include "tc_internal.h"
 
+#define INPUT_OUTPUT_SIZE 10000
+
+static struct compress_header *comp_tc_data;
+static struct compress_header *decomp_tc_data;
+
+static void uninitialize_tc_data()
+{
+	free(comp_tc_data->input_buffer);
+	free(comp_tc_data->output_buffer);
+	free(decomp_tc_data->input_buffer);
+	free(decomp_tc_data->output_buffer);
+	free(comp_tc_data);
+	free(decomp_tc_data);
+}
+
+static int initialize_tc_data()
+{
+	comp_tc_data = (struct compress_header *)malloc(sizeof(struct compress_header));
+	if (comp_tc_data == NULL) 
+		return ERROR;
+		
+	comp_tc_data->input_buffer = (unsigned char *)malloc(INPUT_OUTPUT_SIZE);
+	if (comp_tc_data == NULL) 
+		return ERROR;
```

Comment:
This will cause memory leak.
You should free allocated memory before leaving.

Reply:
i am freeing the memory during assert after returning

Reply:
Let me assume some case.
1. run tc once successfully
2. try to run tc again but failed on this function.
3. then it will free allocated memory by un-init function
4. But some variables have previous data which is freed already. Then it will cause unexpected free of allocated by others.

So, in my opinion, you should free before error returning in this function OR add null initialization at un-init function.